src/main/java/com/archisacademy/jobportal/services/impl/JobAppServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+                jobDto.getCompanyName(), jobDto.getPosition(), jobDto.getStartDate());
+
+        if (jobExists) {
+            LOGGER.log(JobAppMessage.JOB_ALREADY_EXISTS, HttpStatus.CONFLICT);
+            return JobAppMessage.JOB_ALREADY_EXISTS;
+        }
+
+        Job job = jobMapper.toEntity(jobDto);
+        jobRepository.save(job);
+        return JobAppMessage.JOB_CREATED_SUCCESS;
+    }
+
+    @Override
+    @Transactional
+    public String updateJob(Long jobId, JobDto jobDto) {
+
+        Job existingJob = jobRepository.findById(jobId).orElseThrow(() -> {
+            LOGGER.log(JobAppMessage.JOB_NOT_FOUND + jobId, HttpStatus.NOT_FOUND);
+            return null;
+        });
+
+        existingJob.setCompanyName(jobDto.getCompanyName());
+        existingJob.setStartDate(jobDto.getStartDate());
+        existingJob.setEndDate(jobDto.getEndDate());
+        existingJob.setLocation(jobDto.getLocation());
+        existingJob.setLocationType(LocationType.valueOf(jobDto.getLocationType()));
+        existingJob.setPosition(jobDto.getPosition());
+        existingJob.setDescription(jobDto.getDescription());
+
+        jobRepository.save(existingJob);
+        return JobAppMessage.JOB_UPDATED_SUCCESS;
+    }
+
+    @Override
+    @Transactional
+    public String deleteJob(Long jobId) {
+
+        Job job = jobRepository.findById(jobId).orElseThrow(() -> {
+            LOGGER.log(JobAppMessage.JOB_NOT_FOUND + jobId, HttpStatus.NOT_FOUND);
+            return null;
+        });
+
+        List<JobApplicationMapper> jobApplications = jobApplicationMapperRepository.findByJob(job);
+        if (!jobApplications.isEmpty()) {
+            LOGGER.log(JobAppMessage.JOB_HAS_APPLICATIONS, HttpStatus.CONFLICT);
+            return JobAppMessage.JOB_HAS_APPLICATIONS;
+        }
+
+        jobRepository.delete(job);
+        return JobAppMessage.JOB_DELETED_SUCCESS;
+    }
+
+    @Override
+    public JobDto getJobById(Long jobId) {
+        Job job = jobRepository.findById(jobId).orElseThrow(() -> {
+            LOGGER.log(JobAppMessage.JOB_NOT_FOUND + jobId, HttpStatus.NOT_FOUND);
+            return null;
+        });
+
+        return jobMapper.toDto(job);
+    }
+
+    @Override
+    public Page<JobDto> getAllJobs(int pageNo, int pageSize) {
+        Pageable pageable = PageRequest.of(pageNo, pageSize);
+        Page<Job> jobs = jobRepository.findAll(pageable);
+        List<JobDto> jobDtoList = jobs.getContent().stream()
+                .map(jobMapper::toDto)
+                .collect(Collectors.toList());
+        return new PageImpl<>(jobDtoList, pageable, jobs.getTotalElements());
+    }
+
+    @Override
+    @Transactional
+    public String applyToJob(Long jobId, String userUuid) {
+        Job job = jobRepository.findById(jobId).orElseThrow(() -> {
+            LOGGER.log(JobAppMessage.JOB_NOT_FOUND + jobId, HttpStatus.NOT_FOUND);
+            return null;
+        });
+
```

Comment:
JobApplicationMapper needs to be fetched from db if it is null then create new one otherwise use the existing one